tests/pytorch_tests/function_tests/test_hessian_info_weights.py

```diff
@@ -0,0 +1,245 @@
+
+import torch
+from torch.nn import Conv2d, BatchNorm2d, ReLU, Linear, Hardswish
+
+from model_compression_toolkit.core.pytorch.utils import to_torch_tensor
+import numpy as np
+
+from model_compression_toolkit.core.pytorch.default_framework_info import DEFAULT_PYTORCH_INFO
+from model_compression_toolkit.core.pytorch.pytorch_implementation import PytorchImplementation
+from model_compression_toolkit.target_platform_capabilities.tpc_models.imx500_tpc.latest import generate_pytorch_tpc
+from tests.common_tests.helpers.prep_graph_for_func_test import prepare_graph_with_configs
+from tests.pytorch_tests.model_tests.base_pytorch_test import BasePytorchTest
+import model_compression_toolkit.core.common.hessian as hessian_common
+
+"""
+This test checks the model gradients computation
+"""
+
+
+class basic_model(torch.nn.Module):
+    def __init__(self):
+        super(basic_model, self).__init__()
+        self.conv1 = Conv2d(3, 3, kernel_size=1, stride=1)
+        self.bn = BatchNorm2d(3)
+        self.relu = ReLU()
+
+    def forward(self, inp):
+        x = self.conv1(inp)
+        x = self.bn(x)
+        x = self.relu(x)
+        return x
+
+
+class advanced_model(torch.nn.Module):
+    def __init__(self):
+        super(advanced_model, self).__init__()
+        self.conv1 = Conv2d(3, 3, kernel_size=1, stride=1)
+        self.bn1 = BatchNorm2d(3)
+        self.relu1 = ReLU()
+        self.conv2 = Conv2d(3, 3, kernel_size=1, stride=1)
+        self.bn2 = BatchNorm2d(3)
+        self.relu2 = ReLU()
+        self.dense = Linear(32, 7)
+
+    def forward(self, inp):
+        x = self.conv1(inp)
+        x = self.bn1(x)
+        x = self.relu1(x)
+        x = self.conv2(x)
+        x = self.bn2(x)
+        x = self.relu2(x)
+        x = self.dense(x)
+        return x
+
+
+class multiple_outputs_model(torch.nn.Module):
+    def __init__(self):
+        super(multiple_outputs_model, self).__init__()
+        self.conv1 = Conv2d(3, 3, kernel_size=1, stride=1)
+        self.bn1 = BatchNorm2d(3)
+        self.relu1 = ReLU()
+        self.conv2 = Conv2d(3, 3, kernel_size=1, stride=1)
+        self.bn2 = BatchNorm2d(3)
+        self.hswish = Hardswish()
+        self.dense = Linear(32, 7)
+
+    def forward(self, inp):
+        x = self.conv1(inp)
+        x = self.bn1(x)
+        x1 = self.relu1(x)
+        x2 = self.conv2(x1)
+        x2 = self.bn2(x2)
+        x3 = self.hswish(x2)
+        x3 = self.dense(x3)
+        return x1, x2, x3
+
+
+def generate_inputs(inputs_shape):
+    inputs = []
+    for in_shape in inputs_shape:
+        t = torch.randn(*in_shape)
+        t.requires_grad_()
+        inputs.append(t)
+    inputs = to_torch_tensor(inputs)
+    return inputs
+
+
+def get_expected_shape(weights_shape, granularity):
+    if granularity==hessian_common.HessianInfoGranularity.PER_ELEMENT:
+        return weights_shape
+    elif granularity==hessian_common.HessianInfoGranularity.PER_TENSOR:
```

Comment:
I think it's better to output a clear shape like (1,)

Reply:
numpy array of scalar is with shape=()

Reply:
I know. I think it's better to align it with what Keras returns. And in general it will makes the tests and code clearer